shared/utils/TasksProvider/TasksProvider.tsx

```diff
@@ -68,31 +66,19 @@ function TasksProvider({ children }: IProps) {
     setCurrStateIndex(prevHistory.length);
   };
 
-  const updateTask = (data: TTask, taskId: string) => {
-    const prevTask = getTaskFromLocalStorage(taskId);
-    const updatedTask = {
-      id: prevTask.id,
-      title: data.title,
-      description: data.description,
-      dueDate: data.dueDate,
-      priority: data.priority,
-      status: prevTask.status,
-    };
-    updateTaskInLocalStorage(updatedTask);
-    const updatedTasks = getUpdatedTasks();
-
-    const prevHistory = history.slice(0, currStateIndex + 1);
-    setHistory([...prevHistory, [...updatedTasks]]);
-    setCurrStateIndex(prevHistory.length);
-  };
-
-  const markTask = (task: TTask) => {
-    task.status = ETaskStatus.COMPLETED;
-    markTaskComplete(task);
-    const updatedTasks = getUpdatedTasks();
+  const handleUpdateTask = async (data: TUpdateTaskDto, taskId: string) => {
+    try {
+      await updateTask({
+        id: taskId,
+        updatedTask: data,
+      }).unwrap();
+      refetch();
```

Comment:
You don't need to refetch. Look into automated refetching with providesTags and invalidatesTags in [rtk docs](https://redux-toolkit.js.org/rtk-query/usage/automated-refetching).